### PR TITLE
fix: upgrade datastores for node 10

### PR DESCRIPTION
**Pull request by jacobheun**
This fixes an issue where node 10 builds are failing due to old versions of level. They have been upgraded in datastore-level.

This PR also moves the responsibility of shimming leveldown in the browser to `datastore-level`

depends on https://github.com/ipfs/js-datastore-level/pull/7
depends on https://github.com/ipfs/js-datastore-fs/pull/16
depends on https://github.com/dignifiedquire/lock-me/pull/3

resolves ipfs/js-ipfs-repo#167
connects to https://github.com/ipfs/js-ipfs/issues/1347

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,14 +58,12 @@
     "big.js": "^5.0.3",
     "cids": "~0.5.3",
     "datastore-core": "~0.4.0",
-    "datastore-fs": "~0.4.2",
-    "datastore-level": "~0.7.0",
+    "datastore-fs": "github:ipfs/js-datastore-fs#fix/node10",
+    "datastore-level": "github:ipfs/js-datastore-level#fix/node10",
```
Comment (diasdavid) [inline package.json]: Note to self: needs https://github.com/ipfs/js-datastore-level/pull/7

File: package.json [hunk 2]
```diff
@@ -59,14 +58,12 @@
     "big.js": "^5.0.3",
     "cids": "~0.5.3",
     "datastore-core": "~0.4.0",
-    "datastore-fs": "~0.4.2",
-    "datastore-level": "~0.7.0",
+    "datastore-fs": "github:ipfs/js-datastore-fs#fix/node10",
```
Comment (diasdavid) [inline package.json]: This has been merged and released, the dep can be updated.

File: package.json [hunk 3]
```diff
@@ -59,14 +58,12 @@
...
+    "datastore-level": "github:ipfs/js-datastore-level#fix/node10",
     "debug": "^3.1.0",
     "interface-datastore": "~0.4.2",
     "ipfs-block": "~0.7.1",
-    "level-js": "timkuijsten/level.js#idbunwrapper",
-    "leveldown": "^1.7.2",
-    "lock-me": "^1.0.3",
+    "lock-me": "github:jacobheun/lock-me#fix/windows",
```
Comment (diasdavid) [inline package.json]: Can we get this PR up so that @dignifiedquire can review and merge?

Comment (jacobheun) [reply package.json]: We've already gotten that merged and released with lock-me 1.0.4. 

`datastore-level` is the only thing left to be merged, and then I will update all the deps for this. I will get the [feedback](https://github.com/ipfs/js-datastore-level/pull/7#discussion_r191051427) for that resolved so we can get it merged in.

**Comment by diasdavid** [comment]
@jacobheun has this been tested against latest js-ipfs?

**Comment by jacobheun** [comment]
@diasdavid yes, this branch was used against the js-ipfs updates and passing ci tests, https://github.com/ipfs/js-ipfs/pull/1358, as well as locally. It should be ready for release.

**Pull request merged**